web_fix_modules_load/static/src/js/session.js

```diff
@@ -0,0 +1,127 @@
+
+    var Session = require("web.Session");
+    var core = require("web.core");
+    var _t = core._t;
+
+    /**
+     * Override session manager to change how modules' are loaded: use ids instead of names.
+     *
+     */
+    Session.include({
+        _modules_info: function () {
+            return odoo._modules_info;
+        },
+        _module_ids: function (names) {
+            const mod_names = names ? names : this.module_list;
+            const info = this._modules_info();
+            const ids = [];
+            _.each(mod_names, function (name) {
+                if (info[name]) {
+                    ids.push(info[name].id);
+                }
+            });
+            console.debug("Session: load module names", mod_names);
+            return ids;
+        },
+        /**
+         * Full override due to no available hook.
+         * The whole code is taken as-is from Odoo core (comments included).
+         * Only the call to `load_translations` has been modified to use `_module_ids`.
+         * @returns: Promise
+         */
+        load_translations: function () {
+            var lang = this.user_context.lang;
+            /* We need to get the website lang at this level.
+            The only way is to get it is to take the HTML tag lang
+            Without it, we will always send undefined if there is no lang
+            in the user_context. */
+            var html = document.documentElement,
+                htmlLang = html.getAttribute("lang");
+            if (!this.user_context.lang && htmlLang) {
+                lang = htmlLang.replace("-", "_");
+            }
+            return _t.database.load_translations(
+                this,
+                this._module_ids(),
+                lang,
+                this.translationURL
+            );
+        },
+        /**
+         * Full override due to no available hook.
+         * The whole code is taken as-is from Odoo core (comments included).
+         * Only the call to `load_qweb` and `bootstrap_translations` have been modified to use `_module_ids`.
+         * @returns: Promise
+         */
+        session_init: function () {
+            var self = this;
+            var prom = this.session_reload();
+
+            if (this.is_frontend) {
+                return prom.then(function () {
+                    return self.load_translations();
+                });
+            }
+
+            return prom.then(function () {
+                var promise = self.load_qweb(self._module_ids().join(","));
+                if (self.session_is_valid()) {
+                    return promise.then(function () {
+                        return self.load_modules();
+                    });
+                }
+                return Promise.all([
+                    promise,
+                    self
+                        .rpc("/web/webclient/bootstrap_translations", {
+                            mods: self._module_ids(),
+                        })
```

Comment:
Are we forced to send module IDs here? `bootstrap_translations` is a `POST` endpoint (so the module IDs are in the payload, not the header), we should not have the issue there?

Reply:
I'll check, thanks

Reply:
Thinking about it twice: I'd say they should all behave in the same way.

Reply:
Your choice!